churaverse-plugins-server/src/synchroBreakPlugin/synchroBreakPlugin.ts

```diff
@@ -64,13 +94,189 @@ export class SynchroBreakPlugin extends BaseGamePlugin {
+    this.sameTimePlayers = []
+  }
+
+  /**
+   * ターンが開始した際の処理
+   */
+  private readonly nyokkiGameTurnStart = (ev: NyokkiGameTurnStartEvent): void => {
+    const sortedPlayersCoin = this.synchroBreakPluginStore.playersCoinRepository.sortedPlayerCoins()
+    this.networkPluginStore.messageSender.send(new UpdatePlayersCoinMessage({ playersCoin: sortedPlayersCoin }))
+
+    const turnNumber = ev.turnNumber
+    const nyokkiGameTurnStartMessage = new NyokkiTurnStartMessage({ turnNumber })
+    this.networkPluginStore.messageSender.send(nyokkiGameTurnStartMessage)
+  }
+
+  /**
+   * ターンが終了した際の処理
+   */
+  private readonly nyokkiGameTurnEnd = (ev: NyokkiGameTurnEnd): void => {
+    const nyokkiCollection: string[] = this.synchroBreakPluginStore.nyokkiCollection.getAllPlayerId()
+
+    // ニョッキしていないプレイヤーIdを取得
+    const noNyokkiPlayerIds = this.participantIds.filter((playerId) => !nyokkiCollection.includes(playerId))
+    const nyokkiTurnSelect = this.synchroBreakPluginStore.turnSelect
+
+    if (nyokkiTurnSelect === undefined) throw new Error('nyokkiTurnSelect is undefined')
+    const nyokkiGameTurnEndMessage = new NyokkiTurnEndMessage({
+      noNyokkiPlayerIds,
+    })
+    this.networkPluginStore.messageSender.send(nyokkiGameTurnEndMessage)
+
+    this.calculateResultPlayersCoin()
+
+    this.resetTurnCollection()
+  }
+
+  /**
+   * ニョッキコレクションとベットコインレポジトリをリセットする
+   */
+  private resetTurnCollection(): void {
+    this.synchroBreakPluginStore.nyokkiCollection.clear()
+    this.synchroBreakPluginStore.betCoinRepository.clear()
+  }
+
+  /**
+   * プレイヤーのコインを計算する
+   */
+  private calculateResultPlayersCoin(): void {
+    this.participantIds.forEach((playerId) => {
+      const calculatedCoins = this.calculateCoins(playerId)
+      this.synchroBreakPluginStore.playersCoinRepository.change(playerId, calculatedCoins)
+    })
+  }
+
+  /**
+   * プレイヤーのアクションに対して、ベットコインと倍率のコイン数を計算する
+   */
+  private calculateCoins(playerId: string): number {
+    const currentCoins = this.synchroBreakPluginStore.playersCoinRepository.get(playerId)
+    if (currentCoins === undefined) throw new Error('currentCoins is undefined')
+
+    const player: Nyokki | undefined = this.synchroBreakPluginStore.nyokkiCollection.get(playerId)
+    if (player === undefined) return currentCoins
+
+    const playerStatus: NyokkiStatus = player.getNyokkiStatus
+    const betCoins = this.synchroBreakPluginStore.betCoinRepository.get(playerId)
+    const totalPlayerNum = this.participantIds.length
+    const playerOrder: string[] = this.synchroBreakPluginStore.nyokkiCollection.playerOrders()
+    if (betCoins === undefined) throw new Error('betCoins is undefined')
+    // playerStatusが'success'の場合にのみpayoutの増加分の値を返す。それ以外はpayoutを0とする
+    const payout = this.synchroBreakPluginStore.betCoinRepository.calculateMultiplier(
+      betCoins,
+      totalPlayerNum,
+      playerOrder.indexOf(playerId)
+    )
+
+    const gain = playerStatus === 'success' ? payout : 0
```

Comment:
この内容のコメントをつけるなら三項演算子の行につける方が自然ですかね
```suggestion
    const payout = this.synchroBreakPluginStore.betCoinRepository.calculateMultiplier(
      betCoins,
      totalPlayerNum,
      playerOrder.indexOf(playerId)
    )

    // playerStatusが'success'の場合にのみpayoutの増加分の値を返す。それ以外はpayoutを0とする
    const gain = playerStatus === 'success' ? payout : 0
```